packages/xenforo.php

```diff
@@ -34,6 +37,57 @@ class Xenforo extends ExportController {
     protected $folders;
     protected $types;
 
+    /**
+     * Export attachments into vanilla-compatibles names
+     */
+    public function doAttachments() {
+        // Check source folder
+        $this->sourceFolder = $this->param('attachmentpath');
+        if (!is_dir($this->sourceFolder)) {
+            trigger_error("Source attachment folder '{$this->sourceFolder}' does not exist.");
+        }
+
+        // Set up a target folder
+        $this->targetFolder = combinePaths(array($this->sourceFolder, 'xf_attachments'));
+        if (!is_dir($this->targetFolder)) {
+            @$made = mkdir($this->targetFolder, 0777, true);
+            if (!$made) {
+                trigger_error("Target attachment folder '{$this->targetFolder}' could not be created.");
+            }
+        }
+
+        $r = $this->ex->query("
+            select
+                ad.data_id,
+                ad.filename,
+                ad.file_hash
+            from
+                :_attachment a
+            join
+                :_attachment_data ad on a.data_id = ad.data_id
+            where
+                a.content_type = 'post'
+        ");
+
+        echo 'Converting attachments...'.PHP_EOL;
```

Comment:
You have multiple echoes in your package, echoes or prints are never used in porter. Porter has its own `comment()` method.